great_expectations/datasource/fluent/interfaces.py

```diff
@@ -226,14 +227,36 @@ def batch_request_options(self) -> tuple[str, ...]:
         )
 
     def add_batch_config(self, name: str) -> BatchConfig:
+        """Add a BatchConfig to this DataAsset.
+        BatchConfig names must be unique within a DataAsset.
+
+        If the DataAsset is tied to a DataContext, the BatchConfig will be persisted.
+
+        Args:
+            name (str): Name of the new batch config.
+
+        Returns:
+            BatchConfig: The new batch config.
+        """
         batch_config_names = {bc.name for bc in self.batch_configs}
         if name in batch_config_names:
             raise ValueError(
                 f'"{name}" already exists (all existing batch_config names are {", ".join(batch_config_names)})'
             )
 
-        self.__fields_set__.add("batch_configs")
+        # Let mypy know that self.datasource is a Datasource (it is currently bound to MetaDatasource)
+        assert isinstance(self.datasource, Datasource)
+
         batch_config = BatchConfig(name=name)
+        batch_config._data_asset = self
+        self.__fields_set__.add("batch_configs")
+        if self.datasource.data_context:
+            store = self.datasource.data_context._datasource_store
+            key = DataContextVariableKey(resource_name=self.datasource.name)
+            if store.has_key(key=key):
+                batch_config = store.add_batch_config(batch_config)
+                batch_config._data_asset = self
```

Comment:
this doesn't feel quite right on a couple levels.  my current thinking is that this logic live in a `Datasourced method, which is passed to the asset when its instantiated. If the asset doesn't have a `Datasource`, it won't have a callback and we can error here. The `Datasource` has a store and can check to see if its been persisted yet.

Reply:
Playing out what that would look like:
* we'd have an `add_batch_config`, `update_batch_config`, and `delete_batch_config`,  method on the datasource.
* We'd probably add an `is_persisted` method to datasources as well, since it isn't using the callback pattern expectations are using.
* And then within the asset, we'd just say something like:
```
if self.datasource.is_persisted():
   batch_config = self.datasource.add_batch_config(batch_config)
```